frappe/public/js/frappe/form/multi_select_dialog.js

```diff
@@ -35,18 +35,24 @@ frappe.ui.form.MultiSelectDialog = Class.extend({
 		if(!this.date_field) {
 			this.date_field = "transaction_date";
 		}
-		Object.keys(this.setters).forEach(function(setter) {
-			fields.push({
-				fieldtype: me.target.fields_dict[setter].df.fieldtype,
-				label: me.target.fields_dict[setter].df.label,
-				fieldname: setter,
-				options: me.target.fields_dict[setter].df.options,
-				default: me.setters[setter]
-			});
-			if (count++ < Object.keys(me.setters).length) {
-				fields.push({fieldtype: "Column Break"});
+		if($.isArray(this.setters)) {
```

Comment:
Why not transform the `setters` of type object into an array. That way you don't have to write `if else` at three different places.